### PR TITLE
Adds option to treat StructField of StringType as end-to-end opaque json

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

File: src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala [hunk 1]
```diff
@@ -113,17 +113,26 @@ class CosmosDBRowConverter(serializationConfig: SerializationConfig = Serializat
                    json: Document,
                    schema: StructType): Row = {
 
+    val opaqueJsonFields = getOpaqueJsonFiledNames(schema)
+    logTrace(s"recordAsRow ${opaqueJsonFields.mkString(", ")}")
```
Comment (ealsur) [inline src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala]: is this trace needed?

File: src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala [hunk 2]
```diff
@@ -135,24 +144,45 @@ class CosmosDBRowConverter(serializationConfig: SerializationConfig = Serializat
...
+
+    rawJsonFieldNames
+  }
+
   def mapAsRow(
                 json: Map[String, AnyRef],
                 schema: StructType): Row = {
+
+    val opaqueJsonFields = getOpaqueJsonFiledNames(schema)
+    logTrace(s"mapAsRow ${opaqueJsonFields.mkString(", ")}")
```
Comment (ealsur) [inline src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala]: Is this trace needed?

**Review by ealsur — APPROVED** [review]

File: src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala (hunk 1 above)
Comment (FabianMeiswinkel) [reply src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala]: Removed - was helpful while debugging/developing but not needed anymore

File: src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala (hunk 2 above)
Comment (FabianMeiswinkel) [reply src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala]: Removed - was helpful while debugging/developing but not needed anymore

**Pull request merged**